source/fe/fe_poly_tensor.cc

```diff
@@ -981,32 +1094,76 @@ FE_PolyTensor<dim, spacedim>::fill_fe_face_values(
-  // on the neighborhood between two faces.
-  std::fill(fe_data.sign_change.begin(), fe_data.sign_change.end(), 1.0);
-
-  internal::FE_PolyTensor::get_face_sign_change_rt(cell,
-                                                   *this,
-                                                   this->mapping_kind,
-                                                   fe_data.sign_change);
-
-  internal::FE_PolyTensor::get_face_sign_change_nedelec(cell,
-                                                        *this,
-                                                        this->mapping_kind,
-                                                        fe_data.sign_change);
-
-  for (unsigned int i = 0; i < this->n_dofs_per_cell(); ++i)
+  // TODO: The dof_sign_change only affects Nedelec elements and is not the
+  // correct thing on complicated meshes for higher order Nedelec elements.
+  // Something similar to FE_Q should be done to permute dofs and to change the
+  // dof signs. A static way using tables (as done in the RaviartThomas<dim>
+  // class) is preferable.
+  std::fill(fe_data.dof_sign_change.begin(),
+            fe_data.dof_sign_change.end(),
+            1.0);
+  internal::FE_PolyTensor::get_dof_sign_change_nedelec(cell,
+                                                       *this,
+                                                       this->mapping_kind,
+                                                       fe_data.dof_sign_change);
+
+  // TODO: This, similarly to the Nedelec case, is just a legacy function in 2D
+  // and affects only face_dofs of H(div) conformal FEs. It does nothing in 1D.
+  // Also nothing in 3D since we take care of it by using the
+  // adjust_quad_dof_sign_for_face_orientation_table.
+  internal::FE_PolyTensor::get_dof_sign_change_h_div(cell,
+                                                     *this,
+                                                     this->mapping_kind,
+                                                     fe_data.dof_sign_change);
+
+  // What is the first dof_index on a quad?
+  const unsigned int first_quad_index = this->get_first_quad_index();
+  // How many dofs per quad and how many quad dofs do we have at all?
+  const unsigned int n_dofs_per_quad = this->n_dofs_per_quad();
+  const unsigned int n_quad_dofs =
+    n_dofs_per_quad * GeometryInfo<dim>::faces_per_cell;
+
+  for (unsigned int dof_index = 0; dof_index < this->n_dofs_per_cell();
+       ++dof_index)
     {
-      const MappingKind mapping_kind = get_mapping_kind(i);
+      /*
+       * This assumes that the dofs are ordered by first vertices, lines, quads
+       * and volume dofs. Note that in 2D this always gives false.
+       */
+      const bool is_quad_dof =
+        (dim == 2 ? false :
+                    (first_quad_index <= dof_index) &&
+                      (dof_index < first_quad_index + n_quad_dofs));
+
+      unsigned int i        = dof_index;
+      double       dof_sign = 1.0;
```

Comment:
Pretty much all the same comments as I had above for `fill_fe_values()`.

Reply:
done